se.trixon.Mapollage.yml

```diff
@@ -0,0 +1,53 @@
+id: se.trixon.Mapollage
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: start-mapollage
+separate-locales: false
+
+finish-args:
+  - --device=dri
+  - --filesystem=host
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+
+build-options:
+    env:
+        - name=mapollage
+cleanup:
+  - '/mapollage/start-mapollage'    
+  - '/mapollage/se.trixon.Mapollage.desktop'    
+  - '/mapollage/se.trixon.Mapollage.metainfo.xml'    
+
+modules:
+  - name: mapollage
+    buildsystem: simple
+    build-commands:
+      - install -dm755 /app/bin
+      - install -dm755 /app/$name
+      - cp -r * /app/$name/
+      - chmod -R a-s,go+rX,go-w /app/$name
+      - unzip -o -j $name/core/locale/core_$name.jar org/netbeans/core/startup/frame*
+      - install -Dm644 frame512_$name.png /app/share/icons/hicolor/512x512/apps/$FLATPAK_ID.png
+      - install -Dm644 frame256_$name.png /app/share/icons/hicolor/256x256/apps/$FLATPAK_ID.png
+      - install -Dm644 frame48_$name.gif /app/share/icons/hicolor/48x48/apps/$FLATPAK_ID.png
+      - install -Dm644 frame32_$name.gif /app/share/icons/hicolor/32x32/apps/$FLATPAK_ID.png
+      - install -Dm644 frame_$name.gif /app/share/icons/hicolor/16x16/apps/$FLATPAK_ID.png
+      - install -Dm644 $FLATPAK_ID.desktop /app/share/applications/$FLATPAK_ID.desktop
+      - install -Dm644 $FLATPAK_ID.metainfo.xml /app/share/metainfo/$FLATPAK_ID.metainfo.xml
+      - install -m755 start-mapollage /app/bin/start-mapollage
+    sources:
+      - type: archive
+        url: https://github.com/trixon/mapollage/releases/download/v3.0.0/mapollage-3.0.0-linux.zip
```

Comment:
so how does this work on aarch64?

Reply:
It will not. Please advice on how to display Mapollage for only x86_64, for now.

Reply:
https://docs.flathub.org/docs/for-app-authors/maintenance#only-build-on-x86_64

Reply:
Thanks, I have created the flathub.json that I will push once I have GitHub repository access.

Reply:
You need to push it here otherwise build will fail

Reply:
Got confused by

> It assumes your application is already on Flathub, and that you have access rights to its repository.

but it's pushed now.

Reply:
but it's written in Java. why are ou doing this? Isn't java compile once debug everywhere?

Reply:
Doing what exactly, using java?

If you are referring to the zip so is that not just a compiler artifact it's also contains a stripped custom runtime image, extra resources and configuration. 
No need to repeat a work already done.

As a matter of fact I used https://github.com/flathub/org.apache.netbeans as a resource since Mapollage is built upon the same platform.

Reply:
On looking closely, it isn't x86_64 only either. It uses Flathub provided JDK.

Reply:
Thanks for struggling with this but I still don't get the problem.
Mapollage is not for aarch64 since it depends on JavaFX and to my knowledge, JavaFX does not support aarch64 on linux.